src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar.css.twig

```diff
@@ -323,121 +302,158 @@
-        background-color: #f7f7f7;
-
-        background-image: -moz-linear-gradient(225deg, #e4e4e4, #ffffff);
-        background-image: -webkit-gradient(linear, 100% 0%, 0% 0%, from(#e4e4e4), to(#ffffff));
-        background-image: -o-linear-gradient(135deg, #e4e4e4, #ffffff);
-        background: linear-gradient(225deg, #e4e4e4, #ffffff);
-
-        border-radius: 0 0 0 16px;
+        top: 0;
     }
 
     .sf-toolbarreset {
-        background-image: -moz-linear-gradient(225deg, #e4e4e4, #ffffff);
-        background-image: -webkit-gradient(linear, 100% 0%, 0% 0%, from(#e4e4e4), to(#ffffff));
-        background-image: -o-linear-gradient(135deg, #e4e4e4, #ffffff);
-        background: linear-gradient(225deg, #e4e4e4, #ffffff);
-
-        top: 0;
         bottom: auto;
-        border-bottom: 1px solid #bbb;
+        box-shadow: 0 1px 0px rgba(0, 0, 0, 0.2);
+        top: 0;
     }
 
     .sf-toolbar-block .sf-toolbar-info {
-        top: 39px;
         bottom: auto;
-        border-top-width: 0;
-        border-radius: 0 0 4px 4px;
-    }
-
-    .sf-toolbar-block:hover .sf-toolbar-icon {
-        border-top: none;
-        border-bottom: 1px dotted #DDD;
-        margin-top: 0;
-        margin-bottom: -1px;
+        top: 36px;
     }
 {% endif %}
 
 {% if not floatable %}
     .sf-toolbarreset {
         position: static;
-        background: #cbcbcb;
-
-        background-image: -moz-linear-gradient(90deg, #cbcbcb, #e8e8e8) !important;
-        background-image: -webkit-gradient(linear, 0% 0%, 100% 0%, from(#cbcbcb), to(#e8e8e8)) !important;
-        background-image: -o-linear-gradient(180deg, #cbcbcb, #e8e8e8) !important;
-        background: linear-gradient(90deg, #cbcbcb, #e8e8e8) !important;
     }
 {% endif %}
 
-/***** Media query *****/
-@media screen and (max-width: 779px) {
-    .sf-toolbar-block .sf-toolbar-icon > * > :first-child ~ * {
+/* Responsive Design */
+.sf-toolbar-icon .sf-toolbar-label,
+.sf-toolbar-icon .sf-toolbar-value {
+    display: none;
+}
+.sf-toolbar-block-config .sf-toolbar-icon .sf-toolbar-label {
+    display: inline-block;
+}
+
+/* Legacy Design - these styles are maintained to make old panels look
+   a bit better on the new toolbar */
+.sf-toolbar-block .sf-toolbar-info-piece-additional-detail {
+    color: #AAA;
+    font-size: 12px;
+}
+.sf-toolbar-status-green .sf-toolbar-info-piece-additional-detail,
+.sf-toolbar-status-yellow .sf-toolbar-info-piece-additional-detail,
+.sf-toolbar-status-red .sf-toolbar-info-piece-additional-detail {
+    color: #FFF;
+}
+
+@media (min-width: 768px) {
+
+    .sf-toolbar-icon .sf-toolbar-label,
+    .sf-toolbar-icon .sf-toolbar-value {
+        display: inline;
+    }
+
+    .sf-toolbar-block .sf-toolbar-icon img,
+    .sf-toolbar-block .sf-toolbar-icon svg {
+        top: 6px;
+    }
+    .sf-toolbar-block-config:hover .sf-toolbar-info {
+        right: 0;
+    }
+    .sf-toolbar-block-time .sf-toolbar-icon svg,
+    .sf-toolbar-block-memory .sf-toolbar-icon svg {
```

Comment:
rather than hardcoding this handling for the time and memory panel, it would be better to use an agnostic class which could be opted in by any panel wanting a similar behavior (and this behavior would be used by the time and memory ones of course)